testnet/stacks-node/src/nakamoto_node/peer.rs

```diff
@@ -182,8 +182,13 @@ impl PeerThread {
             .parse()
             .unwrap_or_else(|_| panic!("Failed to parse socket: {}", &config.node.rpc_bind));
 
-        net.bind(&p2p_sock, &rpc_sock)
-            .expect("BUG: PeerNetwork could not bind or is already bound");
+        let did_bind = net
```

Comment:
This doesn't sit right with me.  There ought to be only one place where the PeerNetwork is bound to its ports.  If you're instantiating a new `PeerThread` on the transition from epoch 2 to epoch 3, then there ought to be a dedicated constructor for it in which the caller is expected to supply the epoch 2 `PeerThread` instance.